rfc/rfc-2/Entity-key-encoding-for-kv-store-deprecated.md

```diff
@@ -17,13 +17,19 @@
   under the License.
 -->
 
-# RFC-1: Entity Key Encoding design for KV store
+# RFC-1: Entity Key Encoding design for KV store (deprecated)
```

Comment:
Deprecation means something will not receive further development or improvement.
That thing still works, possibly with known defects or flaws.

However, what I'm seeing here is a removal.
I'm wondering if we need to make it crystal clear. This KV store support is gone,
it is removed, it is not simply deprecated.

Reply:
My original purpose is to keep the history even the code is not valid. Maybe we can change to something like "(not valid)" or "(not applicable)", WDYT? @tengqm .

Reply:
I have removed all the code related to the KV store backend, so it can no longer be used. 
Therefore, "removed" is a more appropriate term than "deprecated" considering the impact of this PR merge.

Reply:
I've changed the document like this

<img width="960" alt="image" src="https://github.com/user-attachments/assets/3ed9706a-29cd-4cb4-9e58-bea71c4ea7b8">

Reply:
@pithecuse527 Can you please also mark rfc-3 as "removed" like rfc-2, not just simply remove it? Thanks a lot.

Reply:
No problem. I have revoked the rfc3 document and mark it as "removed"
<img width="1007" alt="image" src="https://github.com/user-attachments/assets/29edfefa-4159-4440-8ae0-788b92c178aa">